Sources/Fluent/Schema/Schema+Creator.swift

```diff
@@ -134,12 +135,34 @@ extension Schema {
         ) {
             fields += Field(
                 name: "\(entity.name)_id",
-                type: .int,
+                type: E.idType?.fieldType ?? .int,
                 optional: optional,
                 unique: unique,
                 default: `default`
             )
         }
+        
+        public func pivotKeys(
+            left: Entity.Type,
+            right: Entity.Type,
+            suffix: String
+        ) {
+            let leftKeyType = left.idType ?? Schema.Field.KeyType.int
+            let rightKeyType = right.idType ?? Schema.Field.KeyType.int
+            
+            let leftName = "\(left.name)\(suffix)"
+            let rightName = "\(right.name)\(suffix)"
+            
+            fields += Field(name: leftName,
+                            type: leftKeyType.fieldType,
+                            optional: false,
+                            unique: false)
+            
+            fields += Field(name: rightName,
+                            type: rightKeyType.fieldType,
```

Comment:
If you mean adding the `_id` suffix, I slightly reordered things around and that gets added through the "suffix" parameter this method receives. The (only) call site is above in Join.swift:

```
+            builder.pivotKeys(left: left, right: right, suffix: "_\(database.driver.idKey)")
```